app/validations/worship-administrative-unit.js

```diff
@@ -28,18 +28,27 @@ export default {
 
 export function getStructuredIdentifierKBOValidations(store) {
   return {
-    localId: validateKBOUniqueness(store),
+    localId: validateKBO(store),
   };
 }
 
-function validateKBOUniqueness(store) {
+function validateKBONumbers(newKboNumber) {
+  if (!newKboNumber.match(/[^$,.\d]/) && newKboNumber.length === 10) {
+    return true;
+  }
+  return {
+    message: 'Vul het (tiencijferige) KBO nummer in.',
+  };
+}
+
+function validateKBO(store) {
   return async (key, newKboNumber, currentKboNumber) => {
```

Comment:
not sure to understand, can you explain why it's not optimal? It seems to cover all the cases

Reply:
The API call isn't needed if the KBO number doesn't have the correct length. At the moment the length of the KBO number is only validated before the API call call if the new value is the same as the old value which doesn't really make sense to me.

If we trust the data in the database it would make sense to return true if the value hasn't changed as it was before and do the length check only if it has actually changed.

If we don't trust the database data the order should be flipped and the check should be done before the "has the value changed" check.

The order I mentioned just seems more logical to me when reading the code.

Reply:
ah yes indeed, api call is not needed. i changed it accordingly